crates/storage/provider/src/traits/stage_checkpoint.rs

```diff
@@ -0,0 +1,13 @@
+use reth_interfaces::Result;
+use reth_primitives::{stage::StageId, BlockNumber, StageCheckpoint};
+
+/// The trait for fetching stage checkpoint related data.
+#[auto_impl::auto_impl(&, Arc)]
+pub trait StageCheckpointProvider: Send + Sync {
+    /// Fetch the checkpoint for the given stage.
+    fn get_stage_checkpoint(&self, id: StageId) -> Result<Option<StageCheckpoint>>;
+
+    /// Fetch the checkpoint for the last stage ([reth_primitives::stage::StageId]).
+    /// Should return `0` if no progress has been made yet.
+    fn get_minimum_stage_checkpoint_block(&self) -> Result<BlockNumber>;
```

Comment:
I don't fully understand the naming at first glance, it fetches the checkpoint for the last stage in the pipeline but references `minimum`, can you clarify in the docs or rename?

Reply:
the last stage is the minimum pipeline progress. just didn't want to leak `pipeline` name into this provider. does `get_last_stage_checkpoint` sound better?

Reply:
yeah, and maybe a note that that checkpoint in particular will always be <= the checkpoint of any other stage?